src/sagemaker/tensorflow/predictor.py

```diff
@@ -101,10 +101,15 @@ def __call__(self, stream, content_type):
             data = stream.read()
         finally:
             stream.close()
-        try:
-            return json_format.Parse(data, tensor_pb2.TensorProto())
-        except json_format.ParseError:
-            return json.loads(data.decode())
+
+        for possible_response in _POSSIBLE_RESPONSES:
+            try:
+                return protobuf_to_dict(json_format.Parse(data, possible_response()))
```

Comment:
Would google.protobuf.json_format.MessageToDict (with json.loads) be suitable here?

Reply:
Good call!

Reply:
I actually tried and won't work. `google.protobuf.json_format.MessageToDict` serializes the values again, which is the bug behavior that we won't to avoid.